runtime/sema/type.go

```diff
@@ -5395,73 +3858,73 @@ func IsSubType(subType Type, superType Type) bool {
 
 	case AnyResourceType:
 		return subType.IsResourceType()
-	}
 
-	switch typedSuperType := superType.(type) {
-	case *NumberType:
-		switch subType.(type) {
-		case *NumberType, *SignedNumberType:
+	case NumberType:
+		switch subType {
+		case NumberType, SignedNumberType:
 			return true
 		}
 
-		return IsSubType(subType, &IntegerType{}) ||
-			IsSubType(subType, &FixedPointType{})
+		return IsSubType(subType, IntegerType) ||
+			IsSubType(subType, FixedPointType)
 
-	case *SignedNumberType:
-		if _, ok := subType.(*SignedNumberType); ok {
+	case SignedNumberType:
+		if subType == SignedNumberType {
 			return true
 		}
 
-		return IsSubType(subType, &SignedIntegerType{}) ||
-			IsSubType(subType, &SignedFixedPointType{})
+		return IsSubType(subType, SignedIntegerType) ||
+			IsSubType(subType, SignedFixedPointType)
 
-	case *IntegerType:
-		switch subType.(type) {
-		case *IntegerType, *SignedIntegerType,
-			*IntType, *UIntType,
-			*Int8Type, *Int16Type, *Int32Type, *Int64Type, *Int128Type, *Int256Type,
-			*UInt8Type, *UInt16Type, *UInt32Type, *UInt64Type, *UInt128Type, *UInt256Type,
-			*Word8Type, *Word16Type, *Word32Type, *Word64Type:
+	case IntegerType:
+		switch subType {
+		case IntegerType, SignedIntegerType,
+			IntType, UIntType,
+			Int8Type, Int16Type, Int32Type, Int64Type, Int128Type, Int256Type,
+			UInt8Type, UInt16Type, UInt32Type, UInt64Type, UInt128Type, UInt256Type,
+			Word8Type, Word16Type, Word32Type, Word64Type:
 
 			return true
 
 		default:
 			return false
 		}
 
-	case *SignedIntegerType:
-		switch subType.(type) {
-		case *SignedIntegerType,
-			*IntType,
-			*Int8Type, *Int16Type, *Int32Type, *Int64Type, *Int128Type, *Int256Type:
+	case SignedIntegerType:
+		switch subType {
+		case SignedIntegerType,
+			IntType,
+			Int8Type, Int16Type, Int32Type, Int64Type, Int128Type, Int256Type:
 
 			return true
 
 		default:
 			return false
 		}
 
-	case *FixedPointType:
-		switch subType.(type) {
-		case *FixedPointType, *SignedFixedPointType,
-			*Fix64Type, *UFix64Type:
+	case FixedPointType:
+		switch subType {
+		case FixedPointType, SignedFixedPointType,
+			Fix64Type, UFix64Type:
 
 			return true
 
 		default:
 			return false
 		}
 
-	case *SignedFixedPointType:
-		switch subType.(type) {
-		case *SignedNumberType, *Fix64Type:
+	case SignedFixedPointType:
+		switch subType {
+		case SignedNumberType, Fix64Type:
```

Comment:
Detected this bug in another PR: 

```suggestion
		case SignedFixedPointType, Fix64Type:
```

Reply:
Added with https://github.com/onflow/cadence/pull/732/commits/21d0cd82bb3a6a5d1f657dc9c521dc4f76f069b1